core-bundle/src/EventListener/CsrfTokenCookieSubscriber.php

```diff
@@ -115,7 +115,7 @@ private function requiresCsrf(Request $request, Response $response): bool
             return true;
         }
 
-        if ($request->hasSession() && $request->getSession()->isStarted()) {
+        if ($request->hasSession() && $request->getSession()->isStarted() && [] !== $request->getSession()->all()) {
```

Comment:
```suggestion
        if ($request->hasSession() && $request->getSession()->isStarted() && $request->getSession()->all()) {
```

iirc we decided on `if ($array)` etc. instead of `if (!empty($array)` or `if ([] !== $array)`, at least for type safe checks (`all()` always returns an array).

Reply:
No, that only applies to nullable values and booleans. If you want to check "is not an empty array", then `[] === $…` is correct.